.github/workflows/release.yml

```diff
@@ -52,20 +48,27 @@ jobs:
         repo_path: meshtastic/Meshtastic-device
         github_token: ${{ secrets.GITHUB_TOKEN }}
 
-    - name: Get version strings from tags
-      run: echo "FIRMWARE_VERSION=$(echo ${{ steps.firmware_latest.outputs.tag_name }}|sed -e 's/^v//')" >> $GITHUB_ENV
+    - name: Get version strings from tag_name
+      run: |
+        echo "CUR_FIRMWARE_VERSION=$(echo ${{ steps.firmware_latest.outputs.tag_name }}|sed -e 's/^v//')" >> $GITHUB_ENV
+        echo "SHORT_FIRMWARE_VERSION=$(echo ${{ steps.firmware_latest.outputs.tag_name }}|sed -e 's/^v//' -e 's/.\{8\}$//')" >> $GITHUB_ENV
+
+    - name: Create curfirmwareversion.xml latest
+      run: |
+        rm ./app/src/main/res/values/curfirmwareversion.xml
+        echo -e "<resources>\n    <string name=\"cur_firmware_version\" translatable=\"false\">${{ env.CUR_FIRMWARE_VERSION }}</string>\n    <string name=\"short_firmware_version\" translatable=\"false\">${{ env.SHORT_FIRMWARE_VERSION }}</string>\n</resources>" > ./app/src/main/res/values/curfirmwareversion.xml
 
     - name: Pull firmware latest
       uses: dsaltares/fetch-gh-release-asset@master
       with:
         repo: meshtastic/Meshtastic-device
-        file: firmware-${{ env.FIRMWARE_VERSION }}.zip
+        file: firmware-${{ env.CUR_FIRMWARE_VERSION }}.zip
         target: firmware.zip
         token: ${{ secrets.GITHUB_TOKEN }}
 
     - name: Unzip firmware into assets/firmware
       run: |
-        rm -f app/src/main/assets/firmware
+        rm -rf app/src/main/assets/firmware
```

Comment:
Is -rf necessary?

Reply:
added -rf to make sure we have nothing else inside `/assets/firmware` but the extracted firmware used for the OTA update.